node/pkg/common/chainlock.go

```diff
@@ -5,21 +5,20 @@ import (
 	"encoding/binary"
 	"encoding/hex"
 	"encoding/json"
-	"errors"
 	"fmt"
+	"math"
 	"time"
 
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/wormhole-foundation/wormhole/sdk/vaa"
 	"go.uber.org/zap"
-
-	"github.com/ethereum/go-ethereum/common"
 )
 
 const HashLength = 32
 const AddressLength = 32
 
 type MessagePublication struct {
-	TxHash    common.Hash // TODO: rename to identifier? on Solana, this isn't actually the tx hash
+	TxID      []byte
```

Comment:
I'm wondering whether `TxID` should be a type of its own with some common interfaces implemented, e.g. `.String()`. It might better encapsulate some of the validation going on in this PR:

* bound checks (length of 64 or 32)
* string-to-hash conversions via something like `func (TxID *id) FromHashString(string s) []byte`, avoiding the need to use `eth_common.HexToHash("0x...").Bytes()`


I understand that might be a bit of an annoying change to make at this point but curious about your thoughts about the merits of this approach longer-term.

Reply:
I'm not sure I see the value of that and I don't think I want to tackle that right now anyway.

Reply:
OK no problem